site-src/geps/gep-851.md

```diff
@@ -0,0 +1,102 @@
+# GEP-851: Allow Multiple Certificate Refs per Gateway Listener
+
+* Issue: [#851](https://github.com/kubernetes-sigs/gateway-api/issues/851)
+* Status: Implementable
+
+## TLDR
+
+Replace `CertificateRef` field with a `CertificateRefs` field in Gateway
+Listeners.
+
+## Goals
+
+* Allow certs to be attached for multiple unique Hostnames for the same
+  Listener. This is most helpful in the following situations:
+  * The hostname on the Listener is empty or unspecified allowing Routes to
+    attach arbitrary hostnames to the Listener with Routes.
+  * The hostname on the Listener is a wildcard and Route owners want to attach
+    multi-level subdomains. For example, `*.example.com` on the Listener and
+    `app.dev.example.com` on a Route.
+* Enable specifying RSA and ECDSA certs on the same Listener.
+* Improve cert upgrades by allowing new and old certs to be attached to the same
```

Comment:
I don't think Envoy can support this (happy to be proven wrong though - not 100% confident): https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/transport_sockets/tls/v3/tls.proto.html?highlight=downstreamtlscontext#envoy-v3-api-msg-extensions-transport-sockets-tls-v3-commontlscontext

' In server contexts, the first RSA certificate is used for clients that only support RSA and the first ECDSA certificate is used for clients that support ECDSA.'

Reply:
Do you know what would happen if one of the attached certs was expired? Do you know if it has any way to prioritize certs that haven't expired?

Reply:
Even if Envoy doesn't support that directly, a control plane can still make a choice about which certificate to send to Envoy (e.g. the one with the longest validity time remaining).